src/App.vue

```diff
@@ -481,46 +530,51 @@ export default {
       this.reasoningInProgress = true;
       this.reasoningResults = {};
 
-      // Prepare request for submission.
-      const query = jQuery.param({
-        jsonld: this.getPhylorefsAndPhylogenyAsOntology(),
-      }).replace(/%20/g, '+');  // $.post will do this automatically,
-                                // but we need to do this here so our
-                                // signature works.
-
-      // Sign it with an X-Hub-Signature.
-      const sign = signer({
-          algorithm: 'sha1',
-          secret: 'undefined' // A pre-arranged secret with the server.
-      });
-      const signature = sign(new Buffer(query));
+      // Make sure that the Reason button is updated before we convert the Phyx
+      // file into JSON-LD.
+      const outerThis = this;
+      Vue.nextTick(function () {
+        // Prepare request for submission.
+        const query = jQuery.param({
+          jsonld: outerThis.getPhylorefsAndPhylogenyAsOntology(),
+        }).replace(/%20/g, '+');  // $.post will do this automatically,
+                                  // but we need to do this here so our
+                                  // signature works.
+
+        // Sign it with an X-Hub-Signature.
+        const sign = signer({
+            algorithm: 'sha1',
+            secret: 'undefined'
+        });
+        const signature = sign(new Buffer(query));
 
-      console.log('Query: ', query);
-      console.log('Signature: ', signature);
+        console.log('Query: ', query);
+        console.log('Signature: ', signature);
 
-      jQuery.post({
-        url: 'https://phyloref.rc.ufl.edu/hooks/reason',
-        data: query,
-        headers: {
-          'X-Hub-Signature': signature,
-        },
-      }).done((data) => {
-        this.reasoningResults = data.phylorefs;
-        // console.log('Data retrieved: ', data);
-      }).fail((jqXHR, textStatus, errorThrown) => {
-        // We can try using the third argument, but it appears to be the
-        // HTTP status (e.g. 'Internal Server Error'). So we default to that,
-        // but look for a better one in the JSON response from the server, if
-        // available.
-        let error = errorThrown;
-        if (has(jqXHR, 'responseJSON') && has(jqXHR.responseJSON, 'error')) {
-          error = jqXHR.responseJSON.error;
-        }
-        if (error === undefined || error === '') error = 'unknown error';
-        alert(`Error occurred on server while reasoning: ${error}`);
-      }).always(() => {
-        // Reset "Reasoning" buttons to their usual state.
-        this.reasoningInProgress = false;
+        jQuery.post({
+          url: 'https://phyloref.rc.ufl.edu/hooks/reason',
```

Comment:
Can we avoid hardcoding this?

Reply:
Fixed in f1f72f4, which will be applied in PR #10 before this PR is merged.